imagenet/main.py

```diff
@@ -292,7 +292,7 @@ def train(train_loader, model, criterion, optimizer, epoch, args):
 
         if args.gpu is not None:
             images = images.cuda(args.gpu, non_blocking=True)
-        if torch.cuda.is_available():
+        if args.gpu is not None:
```

Comment:
Changing this condition will break `nn.DataParallel` because the DataParallel model's output will be on GPU0. This situation is (hopefully) clarified in https://github.com/pytorch/examples/issues/862#issuecomment-1111118888